dashboard/app/models/pd/enrollment.rb

```diff
@@ -221,7 +221,13 @@ def send_exit_survey
 
     return unless should_send_exit_survey?
 
-    Pd::WorkshopMailer.exit_survey(self).deliver_now
+    # Don't send if there's no associated survey
+    return unless exit_survey_url
+
+    return unless (mailer = Pd::WorkshopMailer.exit_survey(self))
+
+    mailer.deliver_now
+    update!(survey_sent_at: Time.zone.now)
   end
```

Comment:
Great solution!! Plus, now it feels more organized. Just a quick question for my understanding, what is the line `return unless (mailer = Pd::WorkshopMailer.exit_survey(self))` doing? I'm a little confused why it's needed.

Reply:
Bethany helped me understand this a few days ago, and I was thrilled I could use the syntax here 🤓 

It's `return unless Pd::WorkshopMailer.exit_survey(self)` but if `Pd::WorkshopMailer.exit_survey(self))` returns something, then set `mailer` to its value.

Reply:
Ohhh that's handy! Thank you :)